### PR TITLE
Benchmark au

### PR DESCRIPTION
**Pull request by LukasHedegaard**
### Added
- Benchmarks for Video-based Human Activity Recognition
- Benchmarks for Skeleton-based Human Activity Recognition
- Benchmarks for Object Detection
- Benchmarks for Object Tracking

### Fixed
- Import error for Human Activity Recognition on devices where `av` cannot be installed.

Authors: @iliiliiliili @negarhdr @LukasHedegaard

### CODE REVIEW

**Comment by LukasHedegaard** [comment]
@negarhdr @iliiliiliili 

We have a few style-errors here. Could you both take a look at those relating to your contributions?

```
Report:
[74](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:74)
/home/runner/work/opendr/opendr/projects/perception/object_tracking_2d/benchmark/benchmark_deep_sort.py:63:5: E122: continuation line missing indentation or outdented
[75](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:75)
/home/runner/work/opendr/opendr/projects/perception/object_tracking_2d/benchmark/benchmark_fair_mot.py:54:1: W293: blank line contains whitespace
[76](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:76)
/home/runner/work/opendr/opendr/projects/perception/facial_expression_recognition/landmark_based_facial_expression_recognition/benchmark/benchmark_pstbln.py:104:1: W391: blank line at end of file
[77](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:77)
/home/runner/work/opendr/opendr/projects/perception/skeleton_based_action_recognition/benchmark/benchmark_stgcn.py:40:69: E127: continuation line over-indented for visual indent
[78](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:78)
/home/runner/work/opendr/opendr/projects/perception/skeleton_based_action_recognition/benchmark/benchmark_stgcn.py:41:69: E127: continuation line over-indented for visual indent
[79](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:79)
/home/runner/work/opendr/opendr/projects/perception/skeleton_based_action_recognition/benchmark/benchmark_stgcn.py:44:69: E127: continuation line over-indented for visual indent
[80](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:80)
/home/runner/work/opendr/opendr/projects/perception/skeleton_based_action_recognition/benchmark/benchmark_stgcn.py:45:69: E127: continuation line over-indented for visual indent
[81](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:81)
/home/runner/work/opendr/opendr/projects/perception/skeleton_based_action_recognition/benchmark/benchmark_stgcn.py:101:1: W391: blank line at end of file
[82](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:82)
/home/runner/work/opendr/opendr/projects/perception/object_detection_3d/benchmark/benchmark_voxel.py:62:1: W293: blank line contains whitespace
[83](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:83)
/home/runner/work/opendr/opendr/projects/perception/object_detection_3d/benchmark/benchmark_voxel.py:90:129: E501: line too long (134 > 128 characters)
[84](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:84)
/home/runner/work/opendr/opendr/src/opendr/perception/object_tracking_2d/deep_sort/object_tracking_2d_deep_sort_learner.py:288:1: W293: blank line contains whitespace
[85](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:85)
/home/runner/work/opendr/opendr/src/opendr/perception/object_tracking_2d/fair_mot/object_tracking_2d_fair_mot_learner.py:380:9: E303: too many blank lines (2)
[86](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:86)
/home/runner/work/opendr/opendr/src/opendr/perception/object_tracking_3d/ab3dmot/object_tracking_3d_ab3dmot_learner.py:52:1: W293: blank line contains whitespace
[87](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:87)
/home/runner/work/opendr/opendr/src/opendr/perception/object_tracking_3d/ab3dmot/object_tracking_3d_ab3dmot_learner.py:133:1: W293: blank line contains whitespace
[88](https://github.com/opendr-eu/opendr/runs/5361961660?check_suite_focus=true#step:4:88)
/home/runner/work/opendr/opendr/src/opendr/perception/object_tracking_3d/ab3dmot/object_tracking_3d_ab3dmot_learner.py:138:1: W293: blank line contains whitespace
```

**Comment by LukasHedegaard** [comment]
There are still som linting errors, @iliiliiliili @negarhdr 
Could you take a look at those pertaining to your code?

**Comment by LukasHedegaard** [comment]
There seem to be some installation issues inherited from the develop branch. 
@ad-daniel , can I persuade you to merge from develop and re-trigger the tests once the issues have been alleviated on develop?

**Comment by ad-daniel** [comment]
Hi, I've made a test branch here https://github.com/opendr-eu/opendr/pull/240 and the tests all passed for me, I'm not sure why on your branch you're having issues. I don't have the impression you've changed anything that would make that test fail. I'll retry a few times and let you know.

Concerning the PR itself, I'm not sure it is useful for the user to have the benchmarking scripts included. The end-results yes definitely should be included in the documentation, but the scripts to obtain it, binaries, txt files, proto files, and so forth I'm not so sure. It bloats the toolkit and the end-user likely won't care.
Alternatively we could create a separate repo where we store it in case we need it again later on, instead of pushing it into the toolkit. What do you think @passalis ?

**Comment by passalis** [comment]
Hi all,

I tend to agree. We could for example create a repo, e.g,. `opendr-misc` under `opendr-eu` to keep there all the resources that are useful for the development, but probably not for the end user.

**Comment by LukasHedegaard** [comment]
Good suggestion @ad-daniel. Binaries, txt files, and proto files definitely have limited value here. The scripts for benchmarking, however, I think make sense to include as these are essentially part of the tests. Also, keep in mind that they are placed under /projects, so they should not be included within pip installations for end-users (which only includes /src).

**Comment by LukasHedegaard** [comment]
Awaiting the merge of #215 , which will hopefully solve the missing libraries unit-test issues.

File: projects/perception/activity_recognition/benchmark/install_on_server.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+conda create --name opendr python=3.8 -y
+conda activate opendr
+conda env config vars set OPENDR_HOME=$PWD
```
[the remark this answers is not in the document]
Comment (LukasHedegaard) [reply projects/perception/activity_recognition/benchmark/install_on_server.sh:5]: It is not essential, but it does provide the exact commands I used to set up the benchmark.
I can remove it if you prefer.

[the remark this answers is not in the document]
Comment (ad-daniel) [reply projects/perception/activity_recognition/benchmark/install_on_server.sh:5]: I see, then perhaps better to keep it

**Comment by ad-daniel** [comment]
I've rerun the CI a couple of times and now it fails consistently with `RuntimeError: Failed to fetch video after 10 retries.`
Locally I have the same issue. Is it supposed to retrieve a video from the FTP or elsewhere? Did something change in that regard?

**Comment by LukasHedegaard** [comment]
The branch now passes all test and should be ready for integration, @passalis @ad-daniel

**Review by ad-daniel — APPROVED** [review]
Looks good to me, thank you!

**Comment by LukasHedegaard** [comment]
We need an additional approval for this. Could one of you take a look, @passalis @omichel?

**Review by omichel — APPROVED** [review]
That looks good to me as well.

**Pull request merged**